### ISSUE TITLE
file_glob breaks because of its return statement

### ISSUE THREAD

**Issue by jvfe** [latchbio/latch#120]
- **Python version**: 3.10.5
- **Latch SDK version**: 1.11.1

I was trying to use file_glob in my workflow, [you can see the code in this branch](https://github.com/jvfe/unhost_latch/blob/breaking_glob/wf/__init__.py), and even though registering went without problems, flyte raised an error when the task with the file_glob ran. Here's the stack trace I gathered from the execution:

<details>
<summary>See stack trace</summary>

```
====================
2022-07-24 13:47:59,759 flytekit ERROR Exception occured when executing task: 'PosixPath' object has no attribute 'decode'
====================
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flytekit/exceptions/scopes.py", line 203, in user_entry_point
    return wrapped(*args, **kwargs)
  File "/root/wf/__init__.py", line 44, in fastp
    return file_glob(f"{sample_name}.*", "latch:///fastp_outputs/")
  File "/usr/local/lib/python3.9/site-packages/latch/types/glob.py", line 53, in file_glob
    return [LatchFile(file, remote_directory + file.name) for file in matched]
  File "/usr/local/lib/python3.9/site-packages/latch/types/glob.py", line 53, in <listcomp>
    return [LatchFile(file, remote_directory + file.name) for file in matched]
  File "/usr/local/lib/python3.9/site-packages/latch/types/file.py", line 57, in __init__
    if _is_valid_url(path) and remote_path is None:
  File "/usr/local/lib/python3.9/site-packages/latch/types/utils.py", line 10, in _is_valid_url
    parsed = urlparse(raw_url)
  File "/usr/local/lib/python3.9/urllib/parse.py", line 392, in urlparse
    url, scheme, _coerce_result = _coerce_args(url, scheme)
  File "/usr/local/lib/python3.9/urllib/parse.py", line 128, in _coerce_args
    return _decode_args(args) + (_encode_result,)
  File "/usr/local/lib/python3.9/urllib/parse.py", line 112, in _decode_args
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
  File "/usr/local/lib/python3.9/urllib/parse.py", line 112, in <genexpr>
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
AttributeError: 'PosixPath' object has no attribute 'decode'


During handling of the above exception, another exception occurred:


Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flytekit/core/base_task.py", line 480, in dispatch_execute
    native_outputs = self.execute(**native_inputs)
  File "/usr/local/lib/python3.9/site-packages/flytekit/core/python_function_task.py", line 161, in execute
    return exception_scopes.user_entry_point(self._task_function)(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/flytekit/exceptions/scopes.py", line 214, in user_entry_point
    raise FlyteScopedUserException(*_exc_info())
flytekit.exceptions.scopes.FlyteScopedUserException: 'PosixPath' object has no attribute 'decode'
====================
2022-07-24 13:47:59,760 flytekit.entrypoint ERROR 
!! Begin User Error Captured by Flyte !!
Traceback (most recent call last):


      File "/usr/local/lib/python3.9/site-packages/flytekit/exceptions/scopes.py", line 203, in user_entry_point
        return wrapped(*args, **kwargs)
      File "/root/wf/__init__.py", line 44, in fastp
        return file_glob(f"{sample_name}.*", "latch:///fastp_outputs/")
      File "/usr/local/lib/python3.9/site-packages/latch/types/glob.py", line 53, in file_glob
        return [LatchFile(file, remote_directory + file.name) for file in matched]
      File "/usr/local/lib/python3.9/site-packages/latch/types/glob.py", line 53, in <listcomp>
        return [LatchFile(file, remote_directory + file.name) for file in matched]
      File "/usr/local/lib/python3.9/site-packages/latch/types/file.py", line 57, in __init__
        if _is_valid_url(path) and remote_path is None:
      File "/usr/local/lib/python3.9/site-packages/latch/types/utils.py", line 10, in _is_valid_url
        parsed = urlparse(raw_url)
      File "/usr/local/lib/python3.9/urllib/parse.py", line 392, in urlparse
        url, scheme, _coerce_result = _coerce_args(url, scheme)
      File "/usr/local/lib/python3.9/urllib/parse.py", line 128, in _coerce_args
        return _decode_args(args) + (_encode_result,)
      File "/usr/local/lib/python3.9/urllib/parse.py", line 112, in _decode_args
        return tuple(x.decode(encoding, errors) if x else '' for x in args)
      File "/usr/local/lib/python3.9/urllib/parse.py", line 112, in <genexpr>
        return tuple(x.decode(encoding, errors) if x else '' for x in args)


Message:


    'PosixPath' object has no attribute 'decode'


User error.
!! End Error Captured by Flyte !!


====================
```
</details>

Apparently, it's happening because [file_glob sends the actual paths to LatchFile in its return statement](https://github.com/latchbio/latch/blob/b47b50b5f47e18b40989e447bc00b9d2463f9a08/latch/types/glob.py#L53), instead of the string representation of each path, so when LatchFile tries to validate the URL it errors out. 

I managed to patch this locally in my workflow by just changing it so `str(file)` is used instead, [as you can see in this other branch](https://github.com/jvfe/unhost_latch/blob/patched_glob/wf/__init__.py).

**Comment by kennyworkman** [comment]
@jvfe can you confirm that `v1.11.2` fixes this for you?

**Comment by jvfe** [comment]
Yep, now it works, thanks!

Fixed by dd6f78e854
